dipy/segment/clustering.py

```diff
@@ -0,0 +1,372 @@
+        will be performed on the size of the clusters instead.
+
+        Parameters
+        ----------
+        other: `ClusterMap` object or int
+            Object to compare to.
+        op: rich comparison operators (see module `operator`)
+            Valid operators are: lt, le, eq, ne, gt or ge.
+
+        Returns
+        -------
+        bool or 1D array (bool)
+            When comparing to another `ClusterMap` object, it returns whether
+            the two `ClusterMap` objects contain the same clusters or not.
+
+            When comparing to an integer the comparison is performed on the
+            clusters sizes, it returns an array of boolean.
+        """
+        if isinstance(other, ClusterMap):
+            if op is operator.eq:
+                return isinstance(other, ClusterMap) \
+                    and len(self) == len(other) \
+                    and self.clusters == other.clusters
+            elif op is operator.ne:
+                return not self == other
+
+            raise NotImplemented("Can only check if two ClusterMap instances are equal or not.")
+
+        elif isinstance(other, int):
+            return np.array([op(len(cluster), other) for cluster in self])
+
+        raise NotImplemented("ClusterMap only supports comparison with a int or another instance of Clustermap.")
+
+    def __eq__(self, other):
+        return self._richcmp(other, operator.eq)
+
+    def __ne__(self, other):
+        return self._richcmp(other, operator.ne)
+
+    def __lt__(self, other):
+        return self._richcmp(other, operator.lt)
+
+    def __le__(self, other):
+        return self._richcmp(other, operator.le)
+
+    def __gt__(self, other):
+        return self._richcmp(other, operator.gt)
+
+    def __ge__(self, other):
+        return self._richcmp(other, operator.ge)
+
+    def add_cluster(self, cluster):
+        """ Adds a new cluster to this cluster map.
+
+        Parameters
+        ----------
+        cluster : `Cluster` object
+            Cluster to be added in this cluster map.
+        """
+        self.clusters.append(cluster)
+        cluster.refdata = self.refdata
+
+    def remove_cluster(self, cluster):
+        """ Remove a cluster from this cluster map.
+
+        Parameters
+        ----------
+        cluster : `Cluster` object
+            Cluster to be removed from this cluster map.
+        """
+        self.clusters.remove(cluster)
+
+    def clear(self):
+        """ Remove all clusters from this cluster map. """
+        self.clusters.clear()
+
+
+class ClusterMapCentroid(ClusterMap):
+    """ Provides functionalities to interact with clustering outputs.
+
+    Useful container to create, remove, retrieve and filter clusters.
+    It allows also to retrieve easely the centroid of every clusters.
+    If `refdata` is given, elements will be returned instead of their
+    index when using `ClusterCentroid` objects.
+
```

Comment:
Docstring lookes like a copy of the super and does not explain current  code. Can you explain why this class is different from ClusterMap?